### ISSUE TITLE
Pre-commit hooks do not run; raises MatchError

### ISSUE THREAD

**Issue by ningyuansg** [source-academy/cadet#45]
On commit, MatchError is raised

```bash
ning:~/github/cadet $ git commit
Pre-commit running...
usage: git stash list [<options>]
   or: git stash show [<stash>]
   or: git stash drop [-q|--quiet] [<stash>]
   or: git stash ( pop | apply ) [--index] [-q|--quiet] [<stash>]
   or: git stash branch <branchname> [<stash>]
   or: git stash [save [--patch] [-k|--[no-]keep-index] [-q|--quiet]
                       [-u|--include-untracked] [-a|--all] [<message>]]
   or: git stash clear
** (MatchError) no match of right hand side value: {"", 1}
    lib/mix/tasks/pre_commit.ex:18: Mix.Tasks.PreCommit.run/1
    (mix) lib/mix/task.ex:314: Mix.Task.run_task/3
    (mix) lib/mix/cli.ex:80: Mix.CLI.run_task/2
    (elixir) lib/code.ex:677: Code.require_file/2
```
_([on asciinema](https://asciinema.org/a/186694))_

- Did try reinstalling dependencies with `mix deps.clean --all` and `mix deps.get`
- Did compile the app (by means of `phx.server`) before commit
- Replicable by @remo5000

**Comment by indocomsoft** [comment]
That's weird -- I can't seem to reproduce the problem.

Could you send me the output of this command?
```bash
uname -a; git --version; elixir --version; erl -eval 'erlang:display(erlang:system_info(otp_release)), halt().'  -noshell
```

For the record, this is the output on my dev machine:
```
Darwin <hostname redacted> 17.6.0 Darwin Kernel Version 17.6.0: Tue May  8 15:22:16 PDT 2018; root:xnu-4570.61.1~1/RELEASE_X86_64 x86_64
git version 2.17.1
Erlang/OTP 20 [erts-9.3.2] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]

Elixir 1.6.5 (compiled with OTP 20)
"20"
```

**Comment by ningyuansg** [comment]
```Linux xps13 4.9.0-6-amd64 #1 SMP Debian 4.9.82-1+deb9u3 (2018-03-02) x86_64 GNU/Linux
git version 2.11.0
Erlang/OTP 19 [erts-8.2.1] [source] [64-bit] [smp:4:4] [async-threads:10] [kernel-poll:false]

Elixir 1.6.0 (compiled with OTP 19)
"19"```

**Comment by ningyuansg** [comment]
My `mix.lock` on a clean, newly cloned repo after `mix deps.get` is different then the on 8e80c02 as  well. Both my `mix.lock` and the diff with master's `mix.lock` is attached. Had to add extension `.txt`, otherwise github wouldn't upload them.

[mix.lock.txt](https://github.com/source-academy/cadet/files/2093043/mix.lock.txt)
[DIFF.txt](https://github.com/source-academy/cadet/files/2093044/DIFF.txt)

**Comment by indocomsoft** [comment]
I have a feeling this is due to `git stash push` not being available in `git` under version 2.13.2

**Comment by ningyuansg** [comment]
Good catch, upgrading git fixed it. Thanks!
